src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AuditApiAction.java

```diff
@@ -0,0 +1,185 @@
+ *     }
+ *   }
+ * }
+ *
+ * PUT _opendistro/_security/api/audit/config
+ * {
+ *   "audit":{
+ *     "enable_rest":true,
+ *     "disabled_rest_categories":[
+ *       "GRANTED_PRIVILEGES",
+ *       "SSL_EXCEPTION"
+ *     ],
+ *     "enable_transport":true,
+ *     "disabled_transport_categories":[
+ *       "GRANTED_PRIVILEGES",
+ *       "AUTHENTICATED"
+ *     ],
+ *     "resolve_bulk_requests":false,
+ *     "log_request_body":true,
+ *     "resolve_indices":true,
+ *     "exclude_sensitive_headers":true,
+ *     "ignore_users":[ ],
+ *     "ignore_requests":[ ]
+ *   },
+ *   "compliance":{
+ *     "internal_config":true,
+ *     "external_config":true,
+ *     "read_metadata_only":true,
+ *     "read_watched_fields":{ },
+ *     "read_ignore_users":[ ],
+ *     "write_metadata_only":true,
+ *     "write_log_diffs":false,
+ *     "write_watched_indices":[ ],
+ *     "write_ignore_users":[ ]
+ *   }
+ * }
+ *
+ * PATCH _opendistro/_security/api/audit
+ * [{"op": "replace", "path": "/config/audit/enable_rest", "value": "true"}]
+ * [{"op": "replace", "path": "/config/compliance/internal_config", "value": "true"}]
+ */
+public class AuditApiAction extends PatchableResourceApiAction {
+    private static final List<Route> routes = ImmutableList.of(
+            new Route(RestRequest.Method.GET, "/_opendistro/_security/api/audit/"),
+            new Route(RestRequest.Method.PUT, "/_opendistro/_security/api/audit/{name}"),
+            new Route(RestRequest.Method.PATCH, "/_opendistro/_security/api/audit/")
+    );
+
+    private static final String RESOURCE_NAME = "config";
+    private final PrivilegesEvaluator privilegesEvaluator;
+    private final ThreadContext threadContext;
+
+    public AuditApiAction(final Settings settings,
+                          final Path configPath,
+                          final RestController controller,
+                          final Client client,
+                          final AdminDNs adminDNs,
+                          final ConfigurationRepository cl,
+                          final ClusterService cs,
+                          final PrincipalExtractor principalExtractor,
+                          final PrivilegesEvaluator privilegesEvaluator,
+                          final ThreadPool threadPool,
+                          final AuditLog auditLog) {
+        super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, privilegesEvaluator, threadPool, auditLog);
+        this.privilegesEvaluator = privilegesEvaluator;
+        this.threadContext = threadPool.getThreadContext();
+    }
+
+    @Override
+    public List<Route> routes() {
+        return routes;
+    }
+
+    @Override
+    protected void handleApiRequest(final RestChannel channel, final RestRequest request, final Client client) throws IOException {
+        // if audit config doc is not available in security index,
+        // disable audit APIs
+        if (!cl.isAuditHotReloadingEnabled()) {
```

Comment:
Yes. We blocking all APIs including GET. Thought that there is no usage for GET as well if the hot-reloading is disabled. However this can be changed. Do you want to return the elasticsearch.yml settings ?